orttraining/orttraining/test/training_ops/cpu/tensor/gather_grad_op_test.cc

```diff
@@ -14,6 +14,54 @@ namespace onnxruntime {
 namespace test {
 
 namespace {
+
+std::pair<std::vector<int64_t>, std::vector<int64_t>> CalculateSortedIndices(const std::vector<int64_t>& indices)
+{
+    std::vector<int64_t> dY_indices_sorted(indices.size());
+    std::iota(dY_indices_sorted.begin(), dY_indices_sorted.end(), 0);
+    sort(dY_indices_sorted.begin(), dY_indices_sorted.end(), [&indices](const size_t& i, const size_t& j) { return indices[i] < indices[j]; } );
```

Comment:
nit: typically we can just pass primitive types like size_t by value.
"std::sort"? not sure why "sort" works here